### PR TITLE
Added keyboards/converter/a1200/mistress1200 variant with atmega16u4 support

### PR DESCRIPTION
**Pull request by 8bits4ever**
Added support for MiSTress 1200 (MiSTer FPGA conversion kit for Amiga 1200 cases). This board uses the Atmega16U4 chip.

Modified config and rules files to get a reduced firmware size.

Minor modification to keymap to support international keys on non-us keyboards.

### CODE REVIEW

File: keyboards/converter/a1200/mistress1200/rules.mk [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# MCU name
+MCU = atmega16u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+EXTRAFLAGS        += -flto
```
Comment (zvecr) [inline keyboards/converter/a1200/mistress1200/rules.mk:6]:
```suggestion
```

File: keyboards/converter/a1200/mistress1200/rules.mk [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite      # Enable Bootmagic Lite
```
Comment (zvecr) [inline keyboards/converter/a1200/mistress1200/rules.mk:11]:
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```

File: keyboards/converter/a1200/mistress1200/rules.mk [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+MOUSEKEY_ENABLE = no        # Mouse keys
+EXTRAKEY_ENABLE = no       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
+NKRO_ENABLE = yes           # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no      # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+AUDIO_ENABLE = no           # Audio output
+COMMAND_ENABLE   = no
```
Comment (zvecr) [inline keyboards/converter/a1200/mistress1200/rules.mk:20]:
```suggestion
LTO_ENABLE = yes
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/converter/a1200/mistress1200/config.h [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+/* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
+#define DEBOUNCE 5
+
+
+/* disable action features */
+//#define NO_ACTION_LAYER
+//#define NO_ACTION_TAPPING
+//#define NO_ACTION_ONESHOT
+#define NO_ACTION_MACRO
+#define NO_ACTION_FUNCTION
```
Comment (drashna) [inline keyboards/converter/a1200/mistress1200/config.h:62]:
```suggestion
```

File: keyboards/converter/a1200/mistress1200/config.h [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+#define DISABLE_LEADER
+
+#ifndef NO_DEBUG
+#define NO_DEBUG
+#endif // !NO_DEBUG
+#if !defined(NO_PRINT) && !defined(CONSOLE_ENABLE)
+#define NO_PRINT
+#endif // !NO_PRINT
```
Comment (drashna) [inline keyboards/converter/a1200/mistress1200/config.h:70]:
```suggestion
```

File: keyboards/converter/a1200/mistress1200/config.h [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+#define DEBOUNCE 5
+
+
+/* disable action features */
+//#define NO_ACTION_LAYER
+//#define NO_ACTION_TAPPING
+//#define NO_ACTION_ONESHOT
+#define NO_ACTION_MACRO
+#define NO_ACTION_FUNCTION
+#define DISABLE_LEADER
```
Comment (drashna) [inline keyboards/converter/a1200/mistress1200/config.h:63]: This doesn't actually do anything. 
```suggestion
```

File: keyboards/converter/a1200/mistress1200/rules.mk [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+MOUSEKEY_ENABLE = no        # Mouse keys
+EXTRAKEY_ENABLE = no       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
+NKRO_ENABLE = yes           # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no      # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+AUDIO_ENABLE = no           # Audio output
+LTO_ENABLE = yes
+UNICODE_ENABLE   = no
```
Comment (drashna) [inline keyboards/converter/a1200/mistress1200/rules.mk:20]:
```suggestion
```

**Review by drashna — COMMENTED** [review]
Is missing a readme, and info.json file.

**Comment by 8bits4ever** [comment]
> Is missing a readme, and info.json file.

Those files are in the main a1200 folder. They are the same for all variants

**Comment by 8bits4ever** [comment]
@drashna 

The modifications I made to keep the size of the firmware below 12288 (to fit inside the atmega16u4) were suggested by you here:
https://github.com/qmk/qmk_firmware/issues/3224

Those still apply?

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**